### ISSUE TITLE
defaults for `storage_options` for the s3 protocol

### ISSUE THREAD

**Issue by keewis** [zarr-developers/VirtualiZarr#222]
- follow-up to the discussion in #135

There are multiple different ways of how to authenticate S3: passing explicit kwargs to `storage_options`, but also environment variables and a session profile (where the credentials are read from a config file, like `.aws/credentials`).

The current defaults in https://github.com/zarr-developers/VirtualiZarr/blob/fdab54cca8d73af4e44903091e5e00bbeb1bb31c/virtualizarr/utils.py#L48 assume we always want to use kwargs for S3, but that will actively prevent other methods from working. For example:
```python
from aiobotocore.session import AioSession

open_virtual_dataset(url, reader_options={"storage_options": {"session": AioSession(profile="myprofile")}})
```
will not use the passed profile because `s3fs` / `aiobotocore` prioritize the kwargs.

The only thing I can think of to fix this is to simply remove `key` and `secret`, but not sure if that's the best option.

**Comment by TomAugspurger** [comment]
IMO, virtualizarr shouldn't have default storage options. I don't think it's in a better position to pick defaults than the underlying library.

**Comment by norlandrhagen** [comment]
Totally agree. [This PR](https://github.com/zarr-developers/VirtualiZarr/pull/228) takes a stab at removing those defaults.

**Comment by keewis** [comment]
should have been closed by #228
